src/app.py

```diff
@@ -120,11 +119,12 @@ def task():
             #completeName = os.path.join(path, filename)
             #file.save(os.path.join(app.root_path, 'static','files',secure_filename(file.filename)))
             database.addFile(newID,filename, file.read())
-            return redirect('/task')
+            return redirect('/task/?taskID='+request.args.get('taskID') )
 
     return render_template("task.html", 
         name=first+' '+last, files=files, 
-        taskObj=database.getTask(int(request.args.get('taskID'))) 
+        taskObj=database.getTask(int(request.args.get('taskID'))),
+        taskID='?taskID='+request.args.get('taskID') 
```

Comment:
The only thing I see is that there is no need for this line. We already send the entire task object to jinja. So if you want get the taskID you can call getTaskID() on taskObj.